libs/ui-watt/src/lib/components/button/watt-button.component.ts

```diff
@@ -17,103 +17,54 @@
 import {
   ChangeDetectionStrategy,
   Component,
-  HostBinding,
-  Injector,
   Input,
-  Type,
   ViewEncapsulation,
 } from '@angular/core';
+import { WattIcon } from '../../foundations/icon';
 
-import { disabledAttributeToken } from './disabled-attribute-token';
-import { typeAttributeToken } from './type-attribute-token';
-import { WattButtonSize } from './watt-button-size';
-import { WattButtonType } from './watt-button-type';
-import { WattButtonVariant } from './watt-button-variant';
-import { WattIcon } from './../../foundations/icon';
-import { WattPrimaryButtonComponent } from './primary-button/watt-primary-button.component';
-import { WattSecondaryButtonComponent } from './secondary-button/watt-secondary-button.component';
-import { WattTextButtonComponent } from './text-button/watt-text-button.component';
+export const WattButtonTypes = [
+  'primary',
+  'secondary',
+  'text',
+  'icon',
+] as const;
+export type WattButtonVariant = typeof WattButtonTypes[number];
+export type WattButtonSize = 'normal' | 'large';
+export type WattButtonType = 'button' | 'reset' | 'submit';
 
 @Component({
   changeDetection: ChangeDetectionStrategy.OnPush,
-  encapsulation: ViewEncapsulation.None,
   selector: 'watt-button',
+  template: `
+    <button
+      mat-button
+      [ngClass]="['watt-button--' + variant, 'watt-button--' + size]"
+      [disabled]="disabled"
+      [type]="type"
+    >
+      <watt-spinner
+        *ngIf="loading"
+        [diameter]="18"
+        class="content-grid-item content-grid-item-spinner"
+      ></watt-spinner>
+      <div class="content-wrapper" [ngClass]="{ hidden: loading }">
```

Comment:
issue: the text and spinner are visible at the same time. Looks like the `hidden` class should be changed to `invisible`.

![button](https://user-images.githubusercontent.com/1096332/167629974-3e21fcb5-653b-4381-8525-da2df37008e8.gif)

Reply:
Fixed